### ISSUE TITLE
Commenting on require_changes flag is not working as expected

### ISSUE THREAD

**Issue by gokmen** [codecov/support#346]
Hi, 

We're currently using codecov in our repository and even though we've enabled commenting on `require_changes`[0][1] it's still commenting on every case [2] 

![koding 2017-02-02 13-48-06](https://cloud.githubusercontent.com/assets/42368/22569951/55079fde-e94e-11e6-872c-e548f18ccedb.png)

[0] https://docs.codecov.io/v4.3.0/docs/pull-request-comments#section-requiring-changes
[1] https://github.com/koding/koding/blob/master/codecov.yml#L7
[2] https://github.com/koding/koding/pull/10283#issuecomment-277088288

**Comment by stevepeak** [comment]
Hey @gokmen Thank you for pointing this out. Looks like your pull changed before I could get a chance to review. My explanation would be:

Two or more files changed in coverage but offset each other resulting in a net 0 change.

The comment still posts on a net 0 change. It would **not post** when, for example, you only adjust files not tracked by Codecov.

**Comment by gokmen** [comment]
Hi @stevepeak thanks for looking out, there are some other examples out there as well. https://github.com/koding/koding/pull/10487 this has no impact on coverage but we are seeing the comment https://github.com/koding/koding/pull/10487#issuecomment-277326753 . But on the other hand in this PR https://github.com/koding/koding/pull/10488 there is no comment, yet :)

Not sure what to do next :|

**Comment by gokmen** [comment]
> The comment still posts on a net 0 change.

is this correct behaviour? if so how can I disable it?

**Comment by stevepeak** [comment]
Hey @gokmen Have you seen this come up again?

Codecov will post the comment in the rare case when coverage is `net 0 change` because of two or more files offsetting each other. If this happens it is important to show the developer which files changed coverage. However, I expect this to be a rare event.

**Comment by gokmen** [comment]
@stevepeak yes it kept happening since we disabled commenting with https://github.com/koding/koding/pull/10513 if you want me to try it out I can re-enable it.

**Comment by stevepeak** [comment]
@gokmen if you could enable again I would appreciate it. I wish to get to the bottom of this quickly if issues persist.

**Comment by gokmen** [comment]
done @stevepeak I'll let you know if it happens again.

**Comment by gokmen** [comment]
hi @stevepeak I'm seeing https://github.com/koding/koding/pull/10660#issuecomment-282977342
